tests/test_acasclient.py

```diff
@@ -291,19 +347,286 @@ def setUp(self):
+            print("Error deleting experiments in tear down: " + str(e))
+
+        try:
+            self.delete_all_cmpd_reg_bulk_load_files(self)
+            print("Successfully deleted all cmpdreg bulk load files")
+        except Exception as e:
+            print("Error deleting bulkloaded files in tear down: " + str(e))
 
-    def tearDown(self):
-        """Tear down test fixtures, if any."""
         try:
-            shutil.rmtree(self.tempdir)
+            self.delete_all_projects(self)
+            print("Successfully deleted all projects (except Global)")
+        except Exception as e:
+            print("Error deleting all projects in tear down: " + str(e))
+
+        try:    
             for username in self.test_usernames:
                 delete_backdoor_user(username)
         finally:
             self.client.close()
 
+    # Helper for testing an experiment upload was successful
+    def experiment_load_test(self, data_file_to_upload, dry_run_mode):
+        response = self.client.\
+            experiment_loader(data_file_to_upload, "bob", dry_run_mode)
+        self.assertIn('results', response)
+        self.assertIn('htmlSummary', response['results'])
+        self.assertIn('errorMessages', response)
+        self.assertIn('hasError', response)
+        self.assertIn('hasWarning', response)
+        self.assertIn('transactionId', response)
+        if dry_run_mode:
+            self.assertIsNone(response['transactionId'])
+        else:
+            self.assertIsNotNone(response['transactionId'])
+        return response
+
+    def delete_all_experiments(self):
+        """ Deletes all experiments """
+        # Currently search is the only way to get all protocols
+        self.basic_experiment_load_code = None
+        protocols = self.client.protocol_search("*")
+        for protocol in protocols:
+            for experiment in protocol["experiments"]:
+                if experiment["ignored"] == False and experiment["deleted"] == False:
+                    self.client.delete_experiment(experiment["codeName"])
+
+            # Verify all experiments are now gone for this protocol
+        all_protocols = self.client.protocol_search("*")
+        not_deleted_experiments = []
+        for protocol in all_protocols:
+            # Loop through all experiments and make sure they are either deleted or ignored
+            for experiment in protocol["experiments"]:
+                if experiment["ignored"] == False and experiment["deleted"] == False:
+                    not_deleted_experiments.append(experiment["codeName"])
+
+        if len(not_deleted_experiments) > 0:
+            raise Exception("Failed to delete all experiments: " + str(not_deleted_experiments))
+                    
+
+    def delete_all_cmpd_reg_bulk_load_files(self):
+        """ Deletes all cmpdreg bulk load files in order by id """
+
+        files = self.client.\
+            get_cmpdreg_bulk_load_files()
+        
+        # sort by id
```

Comment:
This bit is interesting. I would think this code would sort the IDs ascending and then try to delete in the same order the files were loaded.
Wouldn't we want to delete the files in the opposite order from how they were loaded (i.e. remove most recent to prevent there being dependencies)? Am I missing something here?